cmd/dp-deployer/main.go

```diff
@@ -78,7 +82,11 @@ func main() {
 		os.Exit(1)
 	}
 
-	hc := startHealthChecks(ctx, cfg, vc, secretsClient, deploymentsClient)
+	hc, ok := startHealthChecks(ctx, cfg, vc, secretsClient, deploymentsClient, nomadHealthClient)
+	if !ok {
+		log.Event(ctx, "failed to start healthchecks", log.FATAL)
+		os.Exit(1)
+	}
```

Comment:
As per the comment on startHealthChecks(), if that returns an error you can log it here and exit.
```suggestion
	hc, err := startHealthChecks(ctx, cfg, vc, secretsClient, deploymentsClient, nomadHealthClient)
	if err {
		log.Event(ctx, "failed to start healthchecks", log.FATAL, log.Error(err))
		os.Exit(1)
	}
```